### DISCUSSION TITLE
Newline handling

### DISCUSSION THREAD

**Discussion by lilyball** [kdl-org/kdl#27] [Q&A] [answered]
Newlines right now are defined as `('\r' '\n') | '\n'`. I think we should follow the Unicode spec and treat a bare CR as a line separator too. Unicode also recommends that for “a `readline` function” it should stop at CR, LF, CRLF, NEL (U+85), LS (U+2028), PS (U+2029), and even FF (U+0C). This is documented in [§5.8 Newline Guidelines](https://www.unicode.org/versions/Unicode13.0.0/ch05.pdf).

Treating a bare CR as a line separator will simplify parsing. For example, right now kdl-rs will incorrectly parse `”//hello\nworld\r\n` as it reads until the `\r\n` instead of stopping at the `\n`. But I think it will also just make more sense to follow the Unicode standard and treat all of these characters as line terminators.

**Comment by zkat** [comment]
oh hey! This is really interesting! I think this is definitely a change I'd want to make, especially since I'm working on making the language more unicode-friendly (#11)

**Comment by lilyball** [comment, marked as answer]
Fixed in fa60bc63e9dd9806e370be09c45b3224762efe7d and #32.
